pkg/controller/cloudmetrics/cloudmetrics_controller.go

```diff
@@ -77,47 +84,98 @@ var _ reconcile.Reconciler = &ReconcileCloudMetrics{}
+	metricProviderList []providers.MetricsProvider
 }
 
 // Reconcile reads all redis and postgres crs periodically and reconcile metrics for these
 // resources.
 // The Controller will requeue the Request every 5 minutes constantly when RequeueAfter is set
 func (r *ReconcileCloudMetrics) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	r.logger.Info("Reconciling CloudMetrics")
+	ctx := context.TODO()
+	cfgMgr := providers.NewConfigManager(providers.DefaultProviderConfigMapName, request.Namespace, r.client)
 
 	// Fetch all redis crs
 	redisInstances := &integreatlyv1alpha1.RedisList{}
 	err := r.client.List(context.TODO(), redisInstances)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
-	if len(redisInstances.Items) > 0 {
-		for _, redis := range redisInstances.Items {
-			r.logger.Infof("Found redis cr: %s", redis.Name)
-		}
-	} else {
-		r.logger.Info("Found no redis instances")
+	for _, redis := range redisInstances.Items {
+		r.logger.Infof("beginning to reconcile cloud metrics for redis cr: %s", redis.Name)
 	}
 
 	// Fetch all postgres crs
 	postgresInstances := &integreatlyv1alpha1.PostgresList{}
 	err = r.client.List(context.TODO(), postgresInstances)
 	if err != nil {
-		return reconcile.Result{}, err
+		r.logger.Error(err)
 	}
-	if len(postgresInstances.Items) > 0 {
-		for _, postgres := range postgresInstances.Items {
-			r.logger.Infof("Found postgres cr: %s", postgres.Name)
+	for _, postgres := range postgresInstances.Items {
+		r.logger.Infof("beginning to reconcile cloud metrics for postgres cr: %s", postgres.Name)
+		if err := r.reconcilePostgresMetrics(ctx, cfgMgr, &postgres); err != nil {
+			r.logger.Error(err)
 		}
-	} else {
-		r.logger.Info("Found no postgres instances")
 	}
 
-	// Requeue every 5 minutes
+	// we want full control over when we scrape metrics
+	// to allow for this we only have a single requeue
+	// this ensures regardless of errors or return times
+	// all metrics are scraped and exposed at the same time
 	return reconcile.Result{
 		RequeueAfter: watchDuration * time.Second,
 	}, nil
 }
+
+func (r *ReconcileCloudMetrics) reconcilePostgresMetrics(ctx context.Context, cfgMgr providers.ConfigManager, postgres *integreatlyv1alpha1.Postgres) error {
+	stratMap, err := cfgMgr.GetStrategyMappingForDeploymentType(ctx, postgres.Spec.Type)
+	if err != nil {
+		return errorUtil.Wrapf(err, "failed to read deployment type config for deployment %s", postgres.Spec.Type)
+	}
+
+	// Check the CR for existing Strategy
+	strategyToUse := stratMap.Postgres
+	if postgres.Status.Strategy != "" {
+		strategyToUse = postgres.Status.Strategy
+		if strategyToUse != stratMap.Postgres {
+			r.logger.Infof("strategy and provider already set, changing of cloud-resource-config config maps not allowed in existing installation. the existing strategy is '%s' , cloud-resource-config is now set to '%s'. operator will continue to use existing strategy", strategyToUse, stratMap.Postgres)
+		}
+	}
```

Comment:
For metrics do we need to do this? Is checking the CR not enough? If the strategy isn't set in the CR we could just skip it